feat: support for sandbox context customization

Blackman99 opened the pull request:
### Support for customize some sandbox features

* Support customize sandbox head tags, such as `<link ref="stylesheet" href="custom/css/file">`
* Support customize sandbox app import code, such as `import SomePlugin/GlobalComponent from 'customLib'`
* Support customize sandbox app usage code, such as `app.use(SomePlugin)`, `app.component('CompName', GlobalComponent)`

Example for global use [ElementPlus](https://github.com/element-plus/element-plus)

```vue
<script setup>
import { Repl, ReplStore } from '@vue/repl'

const store = new ReplStore()

store.setImportMap({
  imports: {
    'element-plus': 'https://unpkg.com/element-plus@2.2.5/es/index.mjs'
  }
})

const customImportStatements = [`import ElementPlus from 'element-plus'`]
const customAppUsageCodes = [`app.use(ElementPlus)`]
const customHeadTags = [
  `<link ref="stylesheet" href="https://unpkg.com/element-plus@2.2.5/dist/index.css">`
]
</script>
<template>
  <Repl 
    v-bind="{ 
      customImportStatements, 
      customAppUsageCodes, 
      customHeadTags,
      store
    }"
  />
</template>
```

Then in the repl editor, you can use any ElementPlus Components

```vue
<template>
  <el-button>
   A Button
  </el-button>
</template>
```

This PR can possibly solve issue (#41)

yyx990803 reviewed (changes requested):
I like this, but IMO the API needs some adjustments:

1. These should be merged into a single prop `previewOptions`:

```vue
<Repl
  :previewOptions="{
    headHTML: `...`, // inject to end of head
    bodyHTML: `...`, // inject to end of body, e.g. Teleport targets
    customCode: `import foo from 'xxx'; app.use(foo)`
  }"
/>
```

Note there's no need to separate import statements and app code, since import statements are hoisted can be placed anywhere.

The entire options object can also be provided as a single injection instead of three.

Blackman99 commented:
Thanks for api adjustments suggestion. I've changed the origin api to suggested `previewOptions`

But there is still a problem:
If I merge the import code with the app use code (use the `previewOptions.customCode` option), somehow the repl show a error message: Unexpected Identifier which point to the import statement
The error only disappeared when I separate them.

Based on that I've changed the `previewOptions.customCode` to `previewOptions.customCode.importCode` and `previewOptions.customCode.useCode`

Here's my changed commit, in the test/main.ts, has some code I've used to test the feature, using a components lib write by myself: [Casual UI](https://github.com/Blackman99/casual-ui)

fvena commented:
Why hasn't this feature been merged yet? I believe it could be very useful. Is there any default feature that allows us to accomplish this?

fvena commented:
👏👏👏👏👏🎉🎉🍾🍾🎉🎉

bhabgs commented:
如何在编译阶段，就把ui组件给编译了呢？
